compiler/hash-ast/src/tree.rs

```diff
@@ -53,7 +53,7 @@ impl AstVisitor for AstTreeGenerator {
             "entry",
             name.map(|t| TreeNode::branch("name", vec![t]))
                 .into_iter()
-                .chain(ty.map(|t| TreeNode::branch("type", vec![t])).into_iter())
+                .chain(ty.map(|t| TreeNode::branch("type", vec![t])))
```

Comment:
I think that changes like this should be in their own PR in the future. If there is some PR unrelated fmt or clippy change, then I think the policy should be to make a separate PR for it and merge it before your main PR.

Reply:
I've made #883 to fix all the clippy and fmt issues, maybe we should merge this first and then rebase?

Reply:
> I think that changes like this should be in their own PR in the future. If there is some PR unrelated fmt or clippy change, then I think the policy should be to make a separate PR for it and merge it before your main PR.

Yeah sounds reasonable but sometimes can be really annoying..